src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+                elementKind, out adjacentNameAttribute, out adjacentAttributeIsBefore);
+        }
+
+        private class FindAdjacentParameterDocumentationWalker : CSharpSyntaxWalker
+        {
+            private FindAdjacentParameterDocumentationWalker(int parameterIndex, Dictionary<string, int> parameterIndices, XmlNameAttributeElementKind elementKind)
+                : base(SyntaxWalkerDepth.StructuredTrivia)
+            {
+                _parameterIndex = parameterIndex;
+                _parameterIndices = parameterIndices;
+                _elementKind = elementKind;
+            }
+
+            private int _parameterIndex = -1;
+            private Dictionary<string, int> _parameterIndices = new Dictionary<string, int>();
+
+            private XmlNameAttributeSyntax _documentedParameterBefore = null;
+            private int _documentedParameterBeforeIndex = -1;
+            private XmlNameAttributeSyntax _documentedParameterAfter = null;
+            private int _documentedParameterAfterIndex = int.MaxValue;
+
+            private XmlNameAttributeElementKind _elementKind;
+
+            public static void FindAdjacentParameterDocumenteden(string parameterName, IEnumerable<string> allParameterNames,
+                SyntaxToken tokenWithDocumentationComment, XmlNameAttributeElementKind elementKind, out XmlNameAttributeSyntax adjacentNameAttribute, out bool adjacentAttributeIsBefore)
+            {
+                var parameterIndices = new Dictionary<string, int>();
+                int index = 0;
+                foreach (var param in allParameterNames)
+                {
+                    parameterIndices.Add(param, index);
+                    index++;
+                }
+                if (!parameterIndices.ContainsKey(parameterName))
+                {
+                    // something went very wrong
+                    Debug.Assert(false); // TODO message or remove assert
+                    adjacentNameAttribute = null;
+                    adjacentAttributeIsBefore = false;
+                    return;
+                }
+
+                var walker = new FindAdjacentParameterDocumentationWalker(parameterIndices[parameterName], parameterIndices, elementKind);
+                walker.VisitLeadingTrivia(tokenWithDocumentationComment);
+
+                if (walker._documentedParameterBefore != null)
+                {
+                    adjacentNameAttribute = walker._documentedParameterBefore;
+                    adjacentAttributeIsBefore = true;
+                }
+                else
+                {
+                    adjacentNameAttribute = walker._documentedParameterAfter;
+                    adjacentAttributeIsBefore = false;
+                }
+            }
+
+            public override void VisitXmlNameAttribute(XmlNameAttributeSyntax node)
+            {
+                if (node.GetElementKind() == _elementKind)
```

Comment:
@DustinCampbell  This is actually what I did initially. I noticed that the roslyn counts nested and empty tags. In the following example no warnings about missing documentation for parameter are emitted:
``` C#
/// <summary> <param name="arg1"/></param> </summary>
/// <param name="arg2"/> 
static void M(int arg1, int arg2) { }
```
I could just not offer a code fix if no 'adjacent' parameter is found with a simpler method of searching. I just was quite happy to be guaranteed such an adjacent parameter by the warning (I couldn't figure out where to place the new documentation at first) and I tried to capture all the cases. 

In the above example if `arg2` wouldn't be documented it wouldn't be a pretty fix anyways:
``` C#
/// <summary> <param name="arg1"/></param>
/// <param name="arg2"/></param> </summary>
static void M(int arg1, int arg2) { }
```